src/unix/async.c

```diff
@@ -130,14 +134,15 @@ void uv__async_close(uv_async_t* handle) {
 
 
 static void uv__async_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
-  char buf[1024];
-  ssize_t r;
+  assert(w == &loop->async_io_watcher);
+
   struct uv__queue queue;
   struct uv__queue* q;
   uv_async_t* h;
   _Atomic int *pending;
-
-  assert(w == &loop->async_io_watcher);
+#ifndef EVFILT_USER
```

Comment:
This should probably be defined somewhere central (e.g. in src/unix/internal.h) because right now the ifdef guards are a mix of:

- if EVFILT_USER
- if EVFILT_USER && NOTE_TRIGGER
- if EVFILT_USER && NOTE_TRIGGER && !__APPLE__

That doesn't look right to me. It's skipping the read loop now but doesn't call kevent() on macos.

Reply:
> That doesn't look right to me. It's skipping the read loop now but doesn't call kevent() on macos.

Actually, we only include <event.h> that defines `EVFILT_USER` and `NOTE_TRIGGER` on freebsd and netbsd (see the top of this file), so this preprocesser directive will fail and go into the read loop on macOS.

Reply:
> This should probably be defined somewhere central

But I'm with you on that, working on it.

Reply:
Done.